ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -2079,6 +2133,19 @@ def shutdown(self):
     def __del__(self):
         self.shutdown()
 
+    @staticmethod
+    @contextmanager
+    def safe_subprocess_manager():
+        tmpdir = tempfile.mkdtemp(prefix="/tmp/")
+        processes = []
+
+        yield processes, tmpdir
+
+        for process in processes:
+            process.wait()
```

Comment:
where do we keep each subprocess call result?

Reply:
better to add a timeout wait here; and let's stop the subprocess if it doesn't return after timeout.

Reply:
@lolyu, since this is a fire and forget mechanism, I initially dont put any call result. I will add to raise Exception if one of the process failed.

Reply:
One concern is that, if no call result is recorded, it will be hard to debug the call failures in the batch.